src/bms/lex/command.rs

```diff
@@ -474,3 +473,114 @@ impl Channel {
 /// A track, or bar, in the score. It must greater than 0, but some scores may include the 0 track.
 #[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
 pub struct Track(pub u32);
+
+/// Pan value for ExWav sound effect.
+/// Range: [-10000, 10000]. -10000 is leftmost, 10000 is rightmost.
+/// Default: 0.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub struct ExWavPan(i64);
+
+impl ExWavPan {
+    /// Creates a new ExWavPan value.
+    /// Returns `None` if the value is out of range [-10000, 10000].
+    pub fn new(value: i64) -> Option<Self> {
+        (-10000..=10000).contains(&value).then_some(Self(value))
+    }
+
+    /// Returns the underlying value.
+    pub fn value(self) -> i64 {
+        self.0
+    }
+
+    /// Returns the default value (0).
+    pub const fn default() -> Self {
+        Self(0)
+    }
+}
+
+impl Default for ExWavPan {
+    fn default() -> Self {
+        Self(0)
+    }
+}
+
+impl From<i64> for ExWavPan {
+    fn from(value: i64) -> Self {
+        Self::new(value).unwrap_or_else(|| {
+            // Clamp to valid range
+            let clamped = value.clamp(-10000, 10000);
+            Self(clamped)
+        })
+    }
+}
```

Comment:
It doesn't seem that is a good idea about implementing `From` by clamping value. A value to be converted may be a unintended one, so you should impl `TryFrom` instead of it. Please do the same for other `impl From`s.